File: random_tests/probabilities.py
import random


def calculate_probabilities(rounds, dices):
    def compute_shots(n, shots):
        if n == 1:
            return shots

        shot = [random.randint(1, 6) for i in range(dices)]
        total = sum(shot)

        previous_shots = shots.get(str(total), [])[:]
        previous_shots.append(shot)
        shots.update({str(total): previous_shots})

        return compute_shots(n - 1, shots)

    shots = compute_shots(rounds, {})

    probabilities = {key: len(value) / rounds for key, value in shots.items()}

    return probabilities

if __name__ == '__main__':
    rounds = int(input('how many rounds: '))
    dices = int(input('how many dices: '))
    probabilities = calculate_probabilities(rounds, dices)
    print(probabilities)

File: random_tests/probabilities.js
'use strict';


function calculateProbabilities(rounds, dices) {

  function computeShots(n, shots) {
    if(n === 1) {
      return shots;
    }

    const shot = Array.from(Array(dices)).map(
      (_) => Math.floor(Math.random() * 6) + 1);

    const total = shot.reduce((a, b) => a + b);

    const newShots = shots.has(total) ?
    shots.set(total, [shot, ...shots.get(total)]) : shots.set(total, [shot]);

    return computeShots(n - 1, newShots);
  }

  const shots = computeShots(rounds, new Map());

  const probabilities = new Map();
  shots.forEach((value, key) => probabilities.set(key, value.length / rounds));

  return probabilities;
}

if(require.main === module) {
  const rounds = parseInt(process.argv[2]);
  const dices = parseInt(process.argv[3]);
  const probabilities = calculateProbabilities(rounds, dices);
  console.log(probabilities);
}
